core/notifications/notifications.go

```diff
@@ -40,17 +65,92 @@ func (n *notificationService) Init(a *app.App) (err error) {
 	}
 	n.notificationStore = NewNotificationStore(db)
 	n.eventSender = app.MustComponent[event.Sender](a)
+	n.spaceCore = app.MustComponent[spacecore.SpaceCoreService](a)
+	n.spaceService = app.MustComponent[space.Service](a)
+	n.picker = app.MustComponent[block.ObjectGetter](a)
 	return nil
 }
 
 func (n *notificationService) Name() (name string) {
 	return CName
 }
 
-func (n *notificationService) CreateAndSendLocal(notification *model.Notification) error {
-	notification.Id = uuid.New().String()
-	notification.CreateTime = time.Now().Unix()
+func (n *notificationService) Run(_ context.Context) (err error) {
+	notificationContext, notificationCancel := context.WithCancel(context.Background())
+	n.notificationCancel = notificationCancel
+	go n.loadNotificationObject(notificationContext)
+	return nil
+}
 
+func (n *notificationService) indexNotifications(ctx context.Context) {
+	select {
+	case <-ctx.Done():
+		log.Errorf("failed to index notifications: %v", ctx.Err())
+		return
+	default:
+		n.updateNotificationsInLocalStore()
+	}
+}
+
+func (n *notificationService) updateNotificationsInLocalStore() {
+	var notifications map[string]*model.Notification
+	err := block.Do(n.picker, n.notificationId, func(sb smartblock.SmartBlock) error {
+		s := sb.NewState()
+		notifications = s.ListNotifications()
+		return nil
+	})
+	if err != nil {
+		log.Errorf("failed to get notifications from object: %s", err)
+	}
+	lastNotificationTimestamp := make(map[string]int64, 0)
+	for _, notification := range notifications {
+		err := n.notificationStore.SaveNotification(notification)
+		if err != nil {
+			log.Errorf("failed to save notification %s: %s", notification.Id, err)
+		}
+		if notification.AclHeadId != "" && notification.GetCreateTime() > lastNotificationTimestamp[notification.AclHeadId] {
+			n.Lock()
+			n.lastNotificationIdToAcl[notification.AclHeadId] = notification.Id
+			n.Unlock()
+			lastNotificationTimestamp[notification.AclHeadId] = notification.GetCreateTime()
+
+		}
+	}
+}
+
+func (n *notificationService) Close(_ context.Context) (err error) {
+	if n.notificationCancel != nil {
+		n.notificationCancel()
+	}
+	return nil
+}
+
+func (n *notificationService) CreateAndSend(notification *model.Notification) error {
+	if notification.Id == "" {
```

Comment:
Is this a valid scenario? I mean I would probably just return an error and make choosing an id a responsibility of whoever created the notification